### ISSUE TITLE
JSON ingestion mapping conflict with file format MultiJSON

### ISSUE THREAD

**Issue by YaoC** [Azure/azure-kusto-go#251]
**Describe the bug**

When file format is MultiJSON and ingestion mapping type is JSON, a error is returned: `BadRequest_NoRecordsOrWrongFormat: The input stream produced 0 bytes. This usually means that the input JSON stream was ill formed.` Seems that the IngestionMapping option [set both IngestionMappingType and Format to the same `mappingKind`](https://github.com/Azure/azure-kusto-go/blob/69ccd072cfaa6ef1c66360102b887ad15dbc5c8f/kusto/ingest/file_options.go#L244C1-L245C47). And I can't put FileFormat option at the end because `format and ingestion mapping type must match (hint: using ingestion mapping sets the format automatically)`.

**To Reproduce**

```Golang
options := []ingest.FileOption{
    ingest.FlushImmediately(),
    ingest.FileFormat(ingest.MultiJSON),
    ingest.IngestionMapping(mapping, ingest.JSON),
    ingest.ReportResultToTable(),
  }

result, err := in.FromFile(ctx, dataPath, options...)
if err != nil {
    return fmt.Errorf("failed to ingest data: %w", err)
}
```

**Expected behavior**

Ingestion is successful when file format is MultiJSON and ingestion mapping type is JSON. The v0.5.2 we used before was fine.

**Comment by AsafMah** [comment]
Great catch, will release a fix soon.

**Comment by AsafMah** [comment]
We have released 0.15.3 with a fix for this issue.

If you're using the preview branch the fix will come with the next update.

Let us know if that works.
